datafusion/physical-plan/src/aggregates/group_values/group_column.rs

```diff
@@ -140,18 +90,32 @@ where
     }
 }
 
-impl<T: ArrowPrimitiveType> GroupColumn for PrimitiveGroupValueBuilder<T> {
+impl<T: ArrowPrimitiveType, const NULLABLE: bool> GroupColumn
+    for PrimitiveGroupValueBuilder<T, NULLABLE>
+{
     fn equal_to(&self, lhs_row: usize, array: &ArrayRef, rhs_row: usize) -> bool {
-        self.nulls.is_null(lhs_row) == array.is_null(rhs_row)
+        // Perf: skip null check (by short circuit) if input is not ullable
+        let null_match = if NULLABLE {
```

Comment:
`!NULLABLE || self.nulls.is_null(lhs_row) == array.is_null(rhs_row)` maybe?

Reply:
I actually had that formulation in an earlier version but thought this was clearer. I can use that more concise version if you prefer

Reply:
I don't feel very strong for one over the other, I see why you think this version is clearer.